tests/pytests/integration/modules/test_pillar.py

```diff
@@ -366,3 +367,90 @@ def test_pillar_refresh_pillar_ping(salt_cli, salt_minion, key_pillar):
         val = ret.json
         assert key in val
         assert val[key] is True
+
+
+@slowTest
+def test_pillar_refresh_pillar_scheduler(salt_cli, salt_minion):
+    """
+    Ensure schedule jobs in pillar are only updated when values change.
+    """
+
+    top_sls = """
+        base:
+          '{}':
+            - test_schedule
+        """.format(
+        salt_minion.id
+    )
+
+    test_schedule_sls = """
+        schedule:
+          first_test_ping:
+            function: test.ping
+            run_on_start: True
+            seconds: 3600
+        """
+
+    test_schedule_sls2 = """
+        schedule:
+          first_test_ping:
+            function: test.ping
+            run_on_start: True
+            seconds: 7200
+        """
+
+    with pytest.helpers.temp_pillar_file("top.sls", top_sls):
+        with pytest.helpers.temp_pillar_file("test_schedule.sls", test_schedule_sls):
+            # Calling refresh_pillar to update in-memory pillars
+            salt_cli.run(
+                "saltutil.refresh_pillar", wait=True, minion_tgt=salt_minion.id
```

Comment:
why do you still need to run `time.sleep` if you are passing `wait=True` to refresh_pillar, shouldn't it wait for up to 30 seconds for the event?

Reply:
@Ch3LL The `time.sleep` is to ensure that the job in the scheduler has run, the `wait=True` is waiting for the `refresh_pillar` to finish before returning.  I had tested it without the `time.sleep` and it was sporadically failing because the job hadn't finished running yet.

Reply:
sometimes it helps if i would read the comments hehe thanks for the explanation